File: java/src/main/java/com/pahappa/systems/egosmssdk/v1/models/ApiResponse.java
package com.pahappa.systems.egosmssdk.v1.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class ApiResponse {
    @JsonProperty(required = true, value = "Status")
    private ApiResponseCode status;
    @JsonProperty(value = "Message")
    private String message;
    @JsonProperty(value = "Cost")
    private String cost;
    @JsonProperty(value = "MsgFollowUpUniqueCode")
    private String messageFollowUpCode;
    @JsonProperty(value = "Balance")
    private String balance;
}

File: dart/lib/src/v1/models/api_response.dart
import 'api_response_code.dart';

class ApiResponse {
  ApiResponseCode status;
  String? message;
  double? cost;
  String? messageFollowUpCode;
  String? balance;

  ApiResponse.fromJson(Map<String, dynamic> json)
    : status = json['Status']!.toString().toLowerCase() == "ok"
          ? ApiResponseCode.OK
          : ApiResponseCode.Failed,
      message = json['Message'],
      cost = json['Cost'] != null ? double.tryParse(json['Cost'].toString()) : null,
      messageFollowUpCode = json['MsgFollowUpUniqueCode'],
      balance = json['Balance']?.toString();
}
